File: others/WorksApplication/Worksap Package/workspace/China_Intern/China_Intern_Java/src/jp/co/worksap/intern/entities/region/RegionMstDTOReader.java
package jp.co.worksap.intern.entities.region;

import java.io.IOException;

import jp.co.worksap.intern.constants.Constants;
import jp.co.worksap.intern.entities.AbstractDTOReader;

public class RegionMstDTOReader extends AbstractDTOReader<RegionMstDTO> {
	private static final int COLUMN_INDEX_REGION_ID = 0;
	private static final int COLUMN_INDEX_NAME = 1;
	private static final int COLUMN_INDEX_MANAGER_ID = 2;

	private String fileAddress = Constants.DEFAULT_CSV_FOLDER
			+ "REGION_MST.csv";

	/**
	 * use default file address
	 * 
	 * @throws IOException
	 */
	public RegionMstDTOReader() throws IOException {
		super(RegionMstDTOReader.class.getName());
		super.init();
	}

	/**
	 * use customize file address
	 * 
	 * @param fileAddress
	 * @throws IOException
	 */
	public RegionMstDTOReader(final String fileAddress) throws IOException {
		super(RegionMstDTOReader.class.getName());
		this.fileAddress = fileAddress;
		super.init();
	}

	@Override
	protected String getFileArress() {
		return fileAddress;
	}

	@Override
	protected RegionMstDTO convertArrayToDTO(String[] value) throws IOException {
		Long regionId = Long.valueOf(value[COLUMN_INDEX_REGION_ID]);
		String name = value[COLUMN_INDEX_NAME];
		Long managerId = Long.valueOf(value[COLUMN_INDEX_MANAGER_ID]);

		RegionMstDTO dto = new RegionMstDTO(regionId, name, managerId);
		return dto;
	}
}

File: others/WorksApplication/Worksap Package/workspace/China_Intern/China_Intern_Cpp/src/entities/region/region_mst_dto_reader.cpp
#include "region_mst_dto_reader.h"
#include "region_mst_dto.h"
#include "../../constants/constants.h"
#include "../icsv_master_dto.h"
#include <string>
#include <vector>
#include <cstdlib>
using namespace std;

RegionMstDTOReader::RegionMstDTOReader() {
	this->file_address = kDefaultCsvFolder + "REGION_MST.csv";
	AbstractDTOReader::init();
}

RegionMstDTOReader::RegionMstDTOReader(string &file_address) {
	this->file_address = file_address;
	AbstractDTOReader::init();
}

string RegionMstDTOReader::get_file_address() {
	return this->file_address;
}

ICsvMasterDTO *RegionMstDTOReader::convert_array_to_dto(vector<string> &value) {
	long region_id = atol(value[COLUMN_INDEX_REGION_ID].c_str());
	string name = value[COLUMN_INDEX_NAME];
	long manager_id = atol(value[COLUMN_INDEX_MANAGER_ID].c_str());
	ICsvMasterDTO *obj = new RegionMstDTO(region_id, name, manager_id);
	return obj;
}

RegionMstDTOReader::~RegionMstDTOReader() {

}

